components/movementsensor/imuvectornav/imu.go

```diff
@@ -122,20 +114,23 @@ func newVectorNav(
 		return nil, err
 	}
 
-	speed := *newConf.Speed
-	if speed == 0 {
-		speed = 8000000
+	baudRate := *newConf.BaudRate
```

Comment:
oooof, `newVectorNav` is doing so many things. It's hard to keep up. I think we can break it into five different functions: 
1. Configuration Parsing and Setup 
2. Register Reading 
3. IMU Configuration 
4. Acceleration Bias Compensation 
5. Polling Setup 

And call these five functions above in `newVectorNav`

Reply:
Per this conversation, https://github.com/viamrobotics/rdk/pull/4445#discussion_r1799757422.
Specifically that none of these helpers would be testable I think it makes most sense to keep the constructor structured as is.